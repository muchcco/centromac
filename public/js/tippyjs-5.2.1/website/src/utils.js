export function sortActivePages(edges) {
  return [...edges].sort(
    (a, b) => a.node.frontmatter.index - b.node.frontmatter.index,
  );
}

export const ALL_PLACEMENTS = ['top', 'right', 'bottom', 'left'].reduce(
  (acc, placement) => {
    return acc.concat(placement, `${placement}-start`, `${placement}-end`);
  },
  [],
);

export const EXTRA_ANIMATIONS = [
  'shift-away',
  'shift-toward',
  'scale',
  'perspective',
].reduce((acc, animation) => {
  return acc.concat(animation, `${animation}-subtle`, `${animation}-extreme`);
}, []);
