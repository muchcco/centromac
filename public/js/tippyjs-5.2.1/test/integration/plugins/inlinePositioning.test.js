import {fireEvent} from '@testing-library/dom';
import {h, cleanDocumentBody, setTestDefaultProps} from '../../utils';

import tippy from '../../../src';
import inlinePositioning, {
  getInlineBoundingClientRect,
} from '../../../src/plugins/inlinePositioning';
import inlinePositioningData from './inlinePositioningData';

setTestDefaultProps({plugins: [inlinePositioning]});
jest.useFakeTimers();

afterEach(cleanDocumentBody);

describe('inlinePositioning', () => {
  it('true: sets popperInstance.reference = ReferenceObject onShow', () => {
    const instance = tippy(h(), {
      inlinePositioning: true,
      onShow(instance) {
        expect({}.toString.call(instance.popperInstance.reference)).toBe(
          '[object Object]',
        );
      },
    });

    fireEvent.mouseEnter(instance.reference);
    jest.runAllTimers();
  });

  it('false: does not set instance.popperInstance = ReferenceObject onShow', () => {
    const instance = tippy(h(), {
      inlinePositioning: false,
      onShow(instance) {
        expect(instance.popperInstance.reference).toBe(instance.reference);
      },
    });

    fireEvent.mouseEnter(instance.reference);
    jest.runAllTimers();
  });

  it('resets popperInstance.reference onHidden', () => {
    const instance = tippy(h(), {
      onHidden(instance) {
        expect(instance.popperInstance.reference).toBe(instance.reference);
      },
    });

    fireEvent.mouseEnter(instance.reference);
    jest.runAllTimers();
    instance.hide();
  });
});

describe('getInlineBoundingClientRect', () => {
  it('matches placement snapshots', () => {
    inlinePositioningData.forEach(testData => {
      expect(
        getInlineBoundingClientRect(
          testData.placement,
          testData.boundingRect,
          testData.clientRects,
        ),
      ).toMatchSnapshot();
    });
  });

  it('uses boundingRect if clientRects.length < 2', () => {
    const boundingRect = {};
    const clientRects = [{}];

    expect(
      getInlineBoundingClientRect('top', boundingRect, clientRects),
    ).toEqual(boundingRect);
  });
});
