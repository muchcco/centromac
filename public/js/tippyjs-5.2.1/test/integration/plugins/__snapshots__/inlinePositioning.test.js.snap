// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getInlineBoundingClientRect matches placement snapshots 1`] = `
Object {
  "bottom": 199.5,
  "height": 54,
  "left": 252.90625,
  "right": 431.296875,
  "top": 145.5,
  "width": 178.390625,
}
`;

exports[`getInlineBoundingClientRect matches placement snapshots 2`] = `
Object {
  "bottom": 283.5,
  "height": 54,
  "left": 220,
  "right": 287.3125,
  "top": 229.5,
  "width": 67.3125,
}
`;

exports[`getInlineBoundingClientRect matches placement snapshots 3`] = `
Object {
  "bottom": 283.5,
  "height": 36,
  "left": 220,
  "right": 468.578125,
  "top": 247.5,
  "width": 248.578125,
}
`;

exports[`getInlineBoundingClientRect matches placement snapshots 4`] = `
Object {
  "bottom": 373.5,
  "height": 18,
  "left": 220,
  "right": 468.578125,
  "top": 355.5,
  "width": 248.578125,
}
`;

exports[`getInlineBoundingClientRect matches placement snapshots 5`] = `
Object {
  "bottom": 391.5,
  "height": 54,
  "left": 220,
  "right": 468.578125,
  "top": 337.5,
  "width": 248.578125,
  "x": 220,
  "y": 337.5,
}
`;
